can/interfaces/usb_can_analyzer/usb_can_analyzer.py

```diff
@@ -0,0 +1,276 @@
+                   "the serial module installed!")
+    serial = None
+
+
+class CanAnalyzer(BusABC):
+    """
+    Enable basic can communication over a serial device.
+
+    .. note:: See :meth:`can.interfaces.serial.CanAnalyzer._recv_internal`
+              for some special semantics.
+
+    """
+    BITRATE = {
+               1000000: 0x01,
+               800000: 0x02,
+               500000: 0x03,
+               400000: 0x04,
+               250000: 0x05,
+               200000: 0x06,
+               125000: 0x07,
+               100000: 0x08,
+               50000: 0x09,
+               20000: 0x0A,
+               10000: 0x0B,
+               5000: 0x0C
+              }
+    
+    FRAMETYPE = {
+                 "STD":0x01,
+                 "EXT":0x02
+                }
+
+    OPERATIONMODE = {
+                     "normal":0x00,
+                     "loopback":0x01,
+                     "silent":0x02,
+                     "loopback_and_silent":0x03
+                    }
+
+    def __init__(self, channel, baudrate=2000000, timeout=0.1, rtscts=False,
+                 frame_type='STD', operation_mode='normal', bit_rate=500000,
+                 *args, **kwargs):
+        """
+        :param str channel:
+            The serial device to open. For example "/dev/ttyS1" or
+            "/dev/ttyUSB0" on Linux or "COM1" on Windows systems.
+
+        :param int baudrate:
+            Baud rate of the serial device in bit/s (default 115200).
+
+            .. warning::
+                Some serial port implementations don't care about the baudrate.
+
+        :param float timeout:
+            Timeout for the serial device in seconds (default 0.1).
+
+        :param bool rtscts:
+            turn hardware handshake (RTS/CTS) on and off
+
+        """
+        self.bit_rate = bit_rate
+        self.frame_type = frame_type
+        self.op_mode = operation_mode
+        self.filter_id = bytearray([0x00, 0x00, 0x00, 0x00])
+        self.mask_id = bytearray([0x00, 0x00, 0x00, 0x00])
+        if not channel:
+            raise ValueError("Must specify a serial port.")
+
+        self.channel_info = "Serial interface: " + channel
+        self.ser = serial.Serial(
+            channel, baudrate=baudrate, timeout=timeout, rtscts=rtscts)
+
+        super(CanAnalyzer, self).__init__(channel=channel, *args, **kwargs)
+        self.init_frame
+
+    def shutdown(self):
+        """
+        Close the serial interface.
+        """
+        self.ser.close()
+
+    def init_frame(self, timeout=None):
+
+        byte_msg = bytearray()
+        byte_msg.append(0xAA)     # Frame Start Byte 1
+        byte_msg.append(0x55)     # Frame Start Byte 2
+
+        byte_msg.append(0x12)     # Initialization Message ID
+
+        byte_msg.append(CanAnalyzer.BITRATE[self.bit_rate])  # CAN Baud Rate
+        byte_msg.append(CanAnalyzer.FRAMETYPE[self.frame_type]) 
+
+        for i in range(0, 4):
+            byte_msg.append(self.filter_id[i])
```

Comment:
You can also do like this:

```python
byte_msg.extend(self.filter_id)
```

Same applies at other places.

Reply:
I believe I've completed this.